src/ProfileEndpoints.jl

```diff
@@ -242,6 +245,30 @@ end
 
 end  # if isdefined
 
+###
+### Type Inference
+###
+
+function typeinf_start_endpoint(req::HTTP.Request)
+    Core.Compiler.__set_measure_typeinf(true)
```

Comment:
For the old version, where `clear_and_fetch_timings` isn't defined, i think you also need to call reset here.

What i would probably recommend is that you directly call `SnoopCompileCore.start_deep_timing()` instead, since in https://github.com/timholy/SnoopCompile.jl/pull/313, i've changed those functions to always be defined correctly whether it's the new API or the old one.